### PR TITLE
[HDFS-462] Add readiness check to journal nodes

### PR DESCRIPTION
**Pull request by NimaVaziri**
The deployment plan now waits for the journal nodes to succeed their readiness check before proceeding to deploying the name nodes.

### CODE REVIEW

**Comment by NimaVaziri** [comment]
retest this please

File: frameworks/hdfs/src/main/dist/svc.yml [hunk 1]
```diff
@@ -51,6 +51,22 @@ pods:
           {{/TLS_ENABLED}}
         env:
           JOURNALNODE: true
+        # The existence of the journal-data/hdfs/current/committed-txid file indicates HDFS has already been deployed once
```
Comment (gabrielhartmann) [inline frameworks/hdfs/src/main/dist/svc.yml]: I would extend this comment a bit more future readers.  It only describes the simple `if` case, might want to describe what the `else` clause is doing too.

File: frameworks/hdfs/universe/config.json [hunk 2]
```diff
@@ -47,6 +47,17 @@
           "default": "serial",
           "enum" : ["parallel", "serial"]
         },
+        "journal_readiness_check": {
+          "type": "object",
+          "description": "Readiness check configuration properties for Journal Nodes.",
```
Comment (gabrielhartmann) [inline frameworks/hdfs/universe/config.json]: This description is strange.  Maybe, "Determines whether to enable a readiness check for Journal nodes."

File: frameworks/hdfs/src/main/dist/svc.yml (hunk 1 above)
Comment (NimaVaziri) [inline frameworks/hdfs/src/main/dist/svc.yml]: At the surface it's obvious what it's doing, but yes it'd be helpful if it described _why_ it was doing that specifically.

**Comment by NimaVaziri** [comment]
@gabrielhartmann Addressed comments

**Comment by NimaVaziri** [comment]
@gabrielhartmann rebased on master

**Pull request merged**